src/ima.js

```diff
@@ -447,6 +461,19 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
         this._contentSrc = selectedSource[0].url;
       }
     });
+    this.eventManager.listenOnce(this.player, this.player.Event.STALLED, () => {
```

Comment:
Stalled event can come in any browser, seems a bit too wide to couple this.
I think there are other ways to find out that video data is not loaded, such as duration(?)

Reply:
durationchange fired on chrome as well on some cases, Stalled occurred when the data is not available and it could be fine to load the video tag again if no data available and the ad finished but if duration change and the video tag is ok it could be a problem as I see it.

Reply:
As I didn’t debug the duration was a suggestion.
In any case, stalled event is a standard HTML5 video event and as such coupling the bug handling and the event can’t be done. The event can be fired at any given time and then the seeked and ad progress event conditionals will have wrong meaning.

Reply:
It's not something that the end could feel..
I only force the ad progress to fire once for 5 events.
Seeked, I changed the unlistener for specific handler and not for all and force play in handler - I pretty sure that's always his behavior . It couldn't make the user feel the difference I think.